metricbeat/module/mongodb/dbstats/dbstats.go

```diff
@@ -87,23 +82,23 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 
 		err := db.Run("dbStats", &result)
 		if err != nil {
-			err = errors.Wrapf(err, "Failed to retrieve stats for db %s", dbName)
-			logger.Error(err)
-			continue
+			m.Logger().Error(errors.Wrapf(err, "Failed to retrieve stats for db %s", dbName))
+
 		}
 		data, _ := schema.Apply(result)
-		if reported := reporter.Event(mb.Event{MetricSetFields: data}); !reported {
-			logger.Debug("error reporting event")
-			return
+		reported := reporter.Event(mb.Event{MetricSetFields: data})
+		if !reported {
+			//this means the metricset has closed, no point in trying to log anything further
+			return errors.New("metricset has stopped")
```

Comment:
I think here we should just `return nil` and not report an error document. Stopping a metricset is not an error.